end2end/cypress/integration/Users.spec.js

```diff
@@ -7,10 +9,172 @@ describe('Users', () => {
+    cy.get("form").submit()
+    cy.wait('@updateprofile').should(({ request, response }) => {
+      // values should have changed
+      cy.get('input[name="email"]')
+        .should("have.value", "test@test.com");
+      cy.get('input[name="phone"]')
+        .should("have.value", "8005551212");
+    })
+  });
+
+  // edit preferences
+  it('edit preferences', () => {
+    cy.intercept('POST', `http://localhost:3001/api/v3/users/${thisId}/preferences`).as('preferences')
+
+    cy.visit(`/users/${thisId}/profile`);
+    // wait 1 sec, there should be a better way to do this
+    cy.wait(1000)
+    cy.get(`[data-cy="preferences"]`).click().then(() => {
+      // type lab name
+      cy.get('input[name="lab_name"]')
+        .type("my lab")
+        .should("have.value", "my lab");
+
+      // hit reset - value should reset
+      cy.get(`[data-cy=reset]`).click()
+      cy.get('input[name="lab_name"]')
+        .should("have.value", "");
+
+      // type lab name
+      cy.get('input[name="lab_name"]')
+        .type("my lab")
+        .should("have.value", "my lab");
+      // click toggle
+      cy.get(`[data-cy=privatetoggle]`).click()
+
+      cy.get("form").submit()
+      cy.wait('@preferences').should(({ request, response }) => {
+        // values should have changed
+        cy.get('input[name="lab_name"]')
+          .should("have.value", "my lab");
+      })
+    });
+  });
+
+  // click lab agreement
+  it('lab agreement', () => {
+    cy.intercept('POST', `http://localhost:3001/api/v3/users/${thisId}/agreements/lab_agreement`).as('labagreement')
+
+    cy.visit(`/users/${thisId}/profile`);
+    // wait 1 sec, there should be a better way to do this
+    cy.wait(1000)
+    cy.get(`[data-cy="lab_agreement"]`).click().then(() => {
+      cy.get(`[data-cy="agree"]`).click().then(() => {
+        cy.wait('@labagreement').should(({ request, response }) => {
+          // check that agreement button disappears
+          cy.get(`[data-cy="agree"]`).should('not.exist');
+
+          // check that agreed on appears
+          cy.contains('agreed on')
+        })
+      });
+    });
+  });
+
+  // click aquarium agreement
+  it('aquarium agreement', () => {
+    cy.intercept('POST', `http://localhost:3001/api/v3/users/${thisId}/agreements/aquarium_agreement`).as('aquariumagreement')
+
+    cy.visit(`/users/${thisId}/profile`)
+    // wait 1 sec, there should be a better way to do this
+    cy.wait(1000)
+    cy.get(`[data-cy="aquarium_agreement"]`).click().then(() => {
+      cy.get(`[data-cy="agree"]`).click().then(() => {
+        cy.wait('@aquariumagreement').should(({ request, response }) => {
+          // check that agreement button disappears
+          cy.get(`[data-cy="agree"]`).should('not.exist');
+
+          // check that agreed on appears
+          cy.contains('agreed on')
+        })
+      });
+    });
+  });
 
+//  IMPORTANT: Delete the user manually <or> reset the test DB after each (you cannot delete a user in Aquarium)
```

Comment:
I don't think so. Ben is handling this with the setup he is configuring for us. (so that running the tests in Docker uses the test database and cleans the database)